번역 완료 (#391)
Nextjs-kr/Nextjs.kr: docs/02-app/01-building-your-application/04-styling/02-tailwind-css.mdx

---
title: Tailwind CSS
description: Tailwind CSS를 이용하여 Next.js 애플리케이션을 꾸며보세요.
---

<PagesOnly>

<details open>
  <summary>예시</summary>

- [Tailwind CSS와 함께](https://github.com/vercel/next.js/tree/canary/examples/with-tailwindcss)

</details>

</PagesOnly>

[Tailwind CSS](https://tailwindcss.com/)는 Next.js에서 잘 작동되는 유틸리티 우선의 CSS 프레임워크 입니다.

> <!-- is a utility-first CSS framework that works exceptionally well with Next.js. -->

Reviewer: ```suggestion
```

## Tailwind 설치

> <!-- Install the Tailwind CSS packages and run the `init` command to generate both the `tailwind.config.js` and `postcss.config.js` files: -->

Reviewer: ```suggestion
```

> Tailwind CSS 패키지들을 설치 후 `tailwind.config.js`파일과 `postcss.config.js`파일을 생성하기 위해 `init` 커맨드를 사용합니다.

Reviewer: ```suggestion
Tailwind CSS 패키지들을 설치 후 `tailwind.config.js` 파일과 `postcss.config.js` 파일을 생성하기 위해 `init` 커맨드를 사용합니다.
```

```bash filename="Terminal"
npm install -D tailwindcss postcss autoprefixer
npx tailwindcss init -p
```

## Tailwind 구성하기

> <!--
> Inside `tailwind.config.js`, add paths to the files that will use Tailwind CSS class names: -->

Reviewer: ```suggestion
```

`tailwind.config.js` 파일 안에 Tailwind CSS 클래스 이름을 사용할 파일 경로를 추가합니다.

```js filename="tailwind.config.js"
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './app/**/*.{js,ts,jsx,tsx,mdx}', // `app` 디렉토리를 추가합니다.
    './pages/**/*.{js,ts,jsx,tsx,mdx}',
    './components/**/*.{js,ts,jsx,tsx,mdx}',

>     //또는 `src` 디렉토리를 사용하는 경우에는:

Reviewer: ```suggestion
    // 또는 `src` 디렉토리를 사용하는 경우라면
```

}
```

> <!-- You do not need to modify `postcss.config.js`. -->

Reviewer: ```suggestion
```

`postcss.config.js`를 수정할 필요는 없습니다.

> ## 스타일 삽입하기

Reviewer: ```suggestion
## 스타일 가져오기
```

> <!-- Add the [Tailwind CSS directives](https://tailwindcss.com/docs/functions-and-directives#directives) that Tailwind will use to inject its generated styles to a [Global Stylesheet](/docs/app/building-your-application/styling/css-modules#global-styles) in your application, for example: -->

Reviewer: ```suggestion
```

> Tailwind가 생성된 스타일을 애플리케이션의 [Global Stylesheet](/docs/app/building-your-application/styling/css-modules#global-styles) 에 삽입하기 위해 사용될 [Tailwind CSS directives](https://tailwindcss.com/docs/functions-and-directives#directives)
> 을 추가합니다.
> 예를 들면 :

Reviewer: ```suggestion
[Tailwind CSS 지시어](https://tailwindcss.com/docs/functions-and-directives#directives)를 추가해 Tailwind가 생성한 스타일을 애플리케이션의 [Global Stylesheet](/docs/app/building-your-application/styling/css-modules#global-styles)로 가져옵니다.
```

```css filename="app/globals.css"
@tailwind base;
@tailwind components;
@tailwind utilities;
```

> <!-- Inside the [root layout](/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required) (`app/layout.tsx`), import the `globals.css` stylesheet to apply the styles to every route in your application. -->

Reviewer: ```suggestion
```

> [root layout](/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required) (`app/layout.tsx`) 내에서 `global.css` 스타일 시트를 가져와 애플리케이션의 모든 루트에 스타일을 적용합니다.

Reviewer: ```suggestion
[루트 레이아웃](/docs/app/building-your-application/routing/pages-and-layouts#root-layout-required) (`app/layout.tsx`) 내에서 `global.css` 스타일 시트를 가져와 애플리케이션의 모든 라우트에 스타일을 적용합니다.
```

```tsx filename="app/layout.tsx" switcher
import type { Metadata } from 'next'

> // 이 스타일은 애플리케이션의 모든 루트에 적용됩니다.

Reviewer: ```suggestion
// 이 스타일은 애플리케이션의 모든 라우트에 적용됩니다.
```

```

```jsx filename="app/layout.js" switcher

> // 이 스타일은 애플리케이션의 모든 루트에 적용됩니다.

Reviewer: ```suggestion
// 이 스타일은 애플리케이션의 모든 라우트에 적용됩니다.
```

}
```

## 클래스를 이용하기

> <!-- After installing Tailwind CSS and adding the global styles, you can use Tailwind's utility classes in your application. -->

Reviewer: ```suggestion
```

}
```

> ## 터보 팩과 함께 사용

Reviewer: ```suggestion
## 터보팩과 함께 사용
```

> <!-- As of Next.js 13.1, Tailwind CSS and PostCSS are supported with [Turbopack](https://turbo.build/pack/docs/features/css#tailwind-css). -->

Reviewer: ```suggestion
```

> Next.js 13.1부터 Tailwind CSS 및 PostCSS는 [터보 팩](https://turbo.build/pack/docs/features/css#tailwind-css)에서 지원됩니다.

Reviewer: ```suggestion
Next.js 13.1부터 Tailwind CSS 및 PostCSS는 [터보팩](https://turbo.build/pack/docs/features/css#tailwind-css)에서 지원됩니다.
```
